cypress/support/api/utils/users.js

```diff
@@ -106,3 +106,30 @@ export function getMailsForUser(email, i = 0) {
     }
   });
 }
+
+export function getMailWithGiftCardExport(email, subject, i = 0) {
+  if (i > 5) {
+    throw new Error(`There is no email Gift Card export for user ${email}`);
+  }
+  return cy.mhGetMailsByRecipient(email).should(mails => {
+    if (!mails.length) {
+      cy.wait(3000);
+      getMailWithGiftCardExport(email, subject, i + 1);
+    } else {
+      cy.mhGetMailsBySubject(subject).should(mailsWithSubject => {
+        if (!mailsWithSubject.length) {
+          cy.wait(10000);
+          getMailWithGiftCardExport(email, subject, i + 1);
+        } else {
+          cy.wrap(mailsWithSubject)
+            .mhFirst()
+            .should("not.eq", undefined)
+            .mhGetBody()
+            .then(body => {
+              expect(body).to.contain(".csv");
```

Comment:
The name of the function is self explanatory

Reply:
I still think its better to keep expects in test body, I don't know if it will be obvious to everyone that getMailWithGiftCardExport this function is checking if in the email body there is a .csv file. Additionally, it can be renamed to getMailWithGiftCardExportAndCheckIfHasLinkToCSV. [michalina-graczyk](https://github.com/michalina-graczyk), [szczecha](https://github.com/szczecha)

Reply:
I agree with @karola312 without going deeper in the test this function tells me only that it gets email 
In other tests, we first get an email, and then in the test body do the action or verify if received email is what we expect e.g     
"should change admin email plugin. TC: SALEOR_3602",
We change default email title in the config then get the email and verify in the test body if the title was changed as expected